bloggies_backend/src/models/user.ts

```diff
@@ -76,6 +77,49 @@ export default class User {
     return res.rows[0];
   }
 
+  /** Update an existing user */
+  static async updateUser(id: number, updateData: any) {
+    try {
+      let query = "";
+
+      for (let key in updateData) {
+        query = query + ` ${key} = '${updateData[key]}', `;
+      }
+
+      await db.query(
+        `UPDATE users
+        SET ${query}  
+        WHERE user_id = $1`,
+        [id]);
+    } catch (err) {
+      throw new ExpressError(`Err: ${err}`, 400);
+    }
+  }
+
+  /** Sets membership_status to INACTIVE. Sets membership_end_date 
+   * to CURRENT_TIMESTAMP. via subscription id */
+  static async cancelSubscription(subscriptionId: string) {
+    await db.query(
+      `UPDATE users 
+      SET membership_status = $2, membership_end_date = $3, subscription_id = $4
+      WHERE subscription_id = $1`,
+      [subscriptionId, INACTIVE, Date.now(), null]);
+  }
+
+  /** Sets membership_status to ACTIVE. Sets membership_start_date to CURRENT_TIMESTAMP. 
+   * Sets membership_end_date to one month from CURRENT_TIMESTAMP. via subscription id */
+   static async startSubscription(subscriptionId: string) {
+    let startTimestamp = new Date(); // UTC format
```

Comment:
This function does that same thing that User.updateMembership does for "active" users. Do we actually need it at all? Maybe instead we can call updateMembership with an appStatus of "active" and the user_id in the 'invoice.paid' case in the stripe router.

Reply:
https://stripe.com/docs/billing/subscriptions/webhooks Seems like we can use stripe subscription web hook to keep track and manage active and inactive subscriptions.

Reply:
Do you mean that we don't need to track membership start and end dates ourselves because we can do that through Stripe?

Reply:
Yes and it can be done via web hooks but```
 A few days prior to renewal, your site receives an invoice.upcoming event at the webhook endpoint. You can listen for this event to add extra invoice items to the upcoming invoice.
Your site receives an invoice.paid event.
Your webhook endpoint finds the customer for whom payment was just made.
**Your webhook endpoint updates the customer’s current_period_end timestamp in your database to the appropriate date in the future (plus a day or two for leeway).**
```

Reply:
I created two methods for `.startSubscription` and `.cancelSubscription` because they both would query for the user via a `subscription_id` and the reason why I didn't use the existing `.updateMembership` was because there wasn't a way to retrieve the user's `user_id` in the `webhook endpoint` because what is sent to us in the `webhook endpoint` are certain stripe objects, that has no reference to our user's `user_id` .